slither/tools/read_storage/read_storage.py

```diff
@@ -73,6 +75,26 @@ def log(self) -> str:
     def log(self, log: str) -> None:
         self._log = log
 
+    @property
+    def block(self) -> int:
+        """If the RPC is for a POA network, the first call to get_block fails, so we inject geth_poa_middleware"""
+        if not self._block:
+            try:
+                self._block = self.web3.eth.get_block("latest")["number"]
+            except ExtraDataLengthError:
+                self.web3.middleware_onion.inject(geth_poa_middleware, layer=0)
+                self._block = self.web3.eth.get_block("latest")["number"]
```

Comment:
Since we're calling the setter, should this return None and be `Optional[int]` if it hasn't been set?

Reply:
I was just thinking about this. We might want to default to "latest", since that was the default value before we added the property getter and setter.